lib/iris/plot.py

```diff
@@ -256,17 +274,87 @@ def _invert_yaxis(v_coord, axes=None):
+
+    If discontinuity occurs but data is masked, raise warning
+    If discontinuity occurs and data is NOT masked, raise error
+
+    Args:
+        coords:
+            Array of the bounds of a 2D coord, of shape (X,Y,4)
+        data:
+            data of the the cube we are plotting
+        abs_tol:
+            tolerance when checking the contiguity
+
+    """
+    if transpose:
+        bounds = coord.bounds.T
+    else:
+        bounds = coord.bounds
+
+    both_dirs_contiguous, diffs_along_x, diffs_along_y = \
+        iris.coords._discontinuity_in_2d_bounds(bounds, abs_tol=abs_tol)
+
+    if not both_dirs_contiguous:
+
+        # True where data exists
+        mask_invert = np.logical_not(data.mask)
+
+        # Where a discontinuity occurs the grid will not be created correctly.
+        # This does not matter if the data is masked as this is not plotted.
+        # So for places where data exists (opposite of the mask) AND a
+        # diff exists. If any exist, raise a warning
+        not_masked_at_discontinuity_along_x = np.any(
+            np.logical_and(mask_invert[:, :-1], diffs_along_x))
+
+        not_masked_at_discontinuity_along_y = np.any(
+            np.logical_and(mask_invert[:-1, ], diffs_along_y))
+
+        # If discontinuity occurs but not masked, any grid will be created
+        #  incorrectly, so raise a warning
+        if not_masked_at_discontinuity_along_x or \
+                not_masked_at_discontinuity_along_y:
+            raise ValueError('The bounds of the {} coordinate are not'
+                             ' contiguous and data is not masked where the'
+                             ' discontiguity occurs. Not able to create a'
+                             ' suitable mesh to give to'
+                             ' Matplotlib'.format(coord.name()))
+        else:
+            msg = 'The bounds of the {} coordinate are not contiguous.  ' \
+                  'However, data is masked where the discontiguity occurs ' \
+                  'so plotting anyway.'.format(coord.name())
+            warnings.warn(msg)
+
+
 def _draw_2d_from_bounds(draw_method_name, cube, *args, **kwargs):
     # NB. In the interests of clarity we use "u" and "v" to refer to the
     # horizontal and vertical axes on the matplotlib plot.
     mode = iris.coords.BOUND_MODE
     # Get & remove the coords entry from kwargs.
     coords = kwargs.pop('coords', None)
     if coords is not None:
-        plot_defn = _get_plot_defn_custom_coords_picked(cube, coords, mode)
+        plot_defn = _get_plot_defn_custom_coords_picked(cube, coords, mode,
+                                                        ndims=2)
     else:
         plot_defn = _get_plot_defn(cube, mode, ndims=2)
 
+    two_dim_contig_atol = kwargs.pop('two_dim_coord_contiguity_atol',
+                                     1e-4)
+    for coord in plot_defn.coords:
+        if coord.ndim == 2 and coord.has_bounds():
```

Comment:
coord isn't necessarily a `iris.coord.Coord` object at this point. It could instead be `None` or a string.
So first we need to check that it is a coord. I would suggest putting a line that does:
 `if isinstance(cube, iris.coords.Coord)`, 
before this line, but apparently it is better practice to do 'duck-typing'. So instead you could do a check such as:
'if hasattr(coord, 'has_bounds'):
if hasattr(coord, 'add_aux_coord')`